Heisenberg4269/SegTree.cpp

```diff
@@ -0,0 +1,59 @@
+class SegTree
+{
+public:
+    vector <int> segTree;
+    int sizeOfArray;
+    SegTree(vector <int> arr)
+    {
+        sizeOfArray = arr.size();
+        segTree.resize(2*sizeOfArray);
+        for(int i = 0; i<sizeOfArray; i++)
+        {
+            segTree[sizeOfArray + i] = arr[i];
+        }
+        for(int i = sizeOfArray-1; i>=1; i--)
+        {
+            int leftChild = 2*i;
+            int rightChild = 2*i + 1;
+            segTree[i] = min(segTree[leftChild],segTree[rightChild]);
+        }
+    }
+
+    void update(int index, int value)
+    {
+        index += sizeOfArray;
+        segTree[index] = value;
+        while(index > 1)
+        {
+            index/=2;
+            int leftChild = 2*index;
+            int rightChild = 2*index + 1;
+            segTree[index] = min(segTree[leftChild], segTree[rightChild]);
+        }
+    }
+
+    int rangeMinQuery(int leftIndex, int rightIndex)
+    {
+        leftIndex += sizeOfArray;
+        rightIndex += sizeOfArray;
+
+        int minval = 1e9;
```

Comment:
`const int min_val = 1e9;`. Also I am not sure if using 1e9 is a good idea because it 1e9 is a floating type that gets cast to int in this case.